### ISSUE TITLE
Bug: ST2U messes with asset settings when importing other asset packages

### ISSUE THREAD

**Issue by SharlatanY** [Seanba/SuperTiled2Unity#90]
If I have ST2U in my project, it sometimes messes with asset settings from other packages when I'm importing them. This bug has existed for a long time but I only now linked it to ST2U. I don't know if other asset types are affected as well (jpg, etc?) since I only tested this with *.png files. Here's the problem:

If you have ST2U installed and import another asset package containing *.png (and maybe others?) files, it overrides the settings of those files. Here's an example to reproduce this.

1.  Create a new unity project and import the asset package in this zip file: 
[png_import_bug_demo_asset.zip](https://github.com/Seanba/SuperTiled2Unity/files/3495893/png_import_bug_demo_asset.zip)
2. Have a look at Assets/PNG/waves. The import settings should look like this (correct settings):
![waves_standard_import_settings](https://user-images.githubusercontent.com/12735825/62925341-710cbf80-bdb2-11e9-8969-01badd9d8087.jpg)
3. Delete the _PNG_ folder
4. Import ST2U (tested this with 1.5.6
5. Again, import the asset from this zip: [png_import_bug_demo_asset.zip](https://github.com/Seanba/SuperTiled2Unity/files/3495893/png_import_bug_demo_asset.zip)
6. Have a look at Assets/PNG/waves. The import settings now look like this (incorrect settings):
![waves_settings_when_imported_after_ST2U](https://user-images.githubusercontent.com/12735825/62925461-ad402000-bdb2-11e9-88de-ed8e66484280.jpg)
As you can see, there are a lot of unintended changes.
For example, the texture type, wrap mode, filtering compression level aren't the same anymore.

This is not a huge deal when I import my own assets because I have to go through all the settings anyway and now what they need to be.

Where this gets problematic is with assets from the asset store. Some assets contain a lot of *png files and really depend on the correct settings.
In those cases, I have to create a new project, import the asset there, also import the asset in my main project and then have to go through every single file of the asset, compare the settings and correct them where things unintentionally have changed during import due to the presence of ST2U.

Thanks in advance for fixing this! :)

**Comment by Seanba** [comment]
Hi there, @SharlatanY. I'll be travelling for the next couple of days so won't be able to look at this much for a bit.

SuperTiled2Unity has a post-processor in it (`SuperTexturePostprocessor`) that sets texture import settings on their first import that are friendly to ST2U (at least that's what it's trying to do). That's so we don't end up with tilesets that look blurred or compressed.

But I guess that doesn't handle cases where we're importing a texture from a Unity Package? This is what I'll try when I get some more time but you might want to try yourself in the meantime and see if works for you okay ...

```cs
using System.IO;
using UnityEditor;
using UnityEngine;

namespace SuperTiled2Unity.Editor
{
    public class SuperTexturePostprocessor : AssetPostprocessor
    {
        private void OnPreprocessTexture()
        {
            //if (AssetDatabase.LoadAssetAtPath<Texture2D>(assetPath) == null) // replace this
            if (!File.Exists(AssetDatabase.GetTextMetaFilePathFromAssetPath(assetPath))) // with this
            {
                // The texture is being imported for the first time
                // Give the imported texture better defaults than provided by stock Unity
                TextureImporter textureImporter = this.assetImporter as TextureImporter;
                textureImporter.textureType = TextureImporterType.Sprite;
                //textureImporter.spriteImportMode = SpriteImportMode.Single;
                textureImporter.mipmapEnabled = false;
                textureImporter.filterMode = FilterMode.Point;
                textureImporter.textureCompression = TextureImporterCompression.Uncompressed;
            }
        }
    }
}
```

**Comment by Seanba** [comment]
Good news. I found a bit of time to test further. It turns out there's an easier way to tell if an asset has already been imported (through the `importSettingsMissing` field) and it checks out with my testing.

You can [download the fix with SuperTiled2Unity 1.5.7](https://seanba.itch.io/supertiled2unity/devlog/94860/supertiled2unity-version-157). Thanks again for the bug report.

**Comment by SharlatanY** [comment]
@Seanba Thanks for fixing this so quickly!

The first solution worked for me as well, in case you're interested.
I have to say, the solutions are pretty ingenious. Probably wouldn't have thought of that so quickly when trying to come up with an idea how to check if an asset gets imported via Unity Package!

Thanks again and have a nice day!
